src/tests/ftest/deployment/target_failure.py

```diff
@@ -0,0 +1,417 @@
+            job_num=job_num, results=ior_results, file_name="test_file_3", oclass=oclass,
+            pool=self.pool, container=self.container)
+
+        # Verify that there's no error.
+        self.log.info("----- IOR results 2 -----")
+        self.log.info(ior_results)
+        if not ior_results[job_num][0]:
+            errors.append(
+                "Error found in third IOR run! {}".format(ior_results[job_num][-1]))
+
+        report_errors(test=self, errors=errors)
+
+    def test_target_failure_parallel(self):
+        """Jira ID: DAOS-10001.
+
+        Verifying that failing a target in one pool doesn't affect other pool.
+
+        1. Create 2 pools and a container in each pool.
+        2. Run IOR with oclass S1 on all containers at the same time.
+        3. Exclude one target from self.pool[1] while IOR is running.
+        4. Verify the IOR failed for self.pool[1].
+        5. Verify the IOR worked for self.pool[0].
+        6. Verify that the self.pool[1].container's Health property is UNCLEAN.
+        7. Reintegrate the excluded target.
+        8. Verify that the self.pool[1].container's Health property is HEALTHY.
+        9. Run IOR again.
+        10. Verify that there's no error this time.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=deployment,target_failure
+        :avocado: tags=target_failure_parallel
+        """
+        pool_qty = 2
+        self.container = []
+
+        # 1. Create 2 pools and a container in each pool. Use the actual size value
+        # rather than percentage because percentage doesn't work well with multiple
+        # pools. Also, this test focuses on the isolation rather than creating the largest
+        # pools possible.
+        self.add_pool_qty(quantity=pool_qty, namespace="/run/pool_size_value/*")
+        for i in range(pool_qty):
+            self.container.append(
+                self.get_container(pool=self.pool[i], namespace="/run/container_wo_rf/*"))
+
+        # 2. Run IOR with oclass S1 on all containers at the same time.
+        ior_results = {}
+        oclass = self.params.get("oclass", "/run/object_class/wo_rf/*")
+        threads = []
+
+        for pool_num in range(pool_qty):
+            threads.append(
+                threading.Thread(
+                    target=self.run_ior_report_error,
+                    args=[ior_results, pool_num, "test_file_1", oclass,
+                          self.pool[pool_num], self.container[pool_num]]))
+
+            threads[-1].start()
+
+        # Wait for a few seconds for IOR to start.
+        self.log.info("Waiting 5 sec for IOR to start writing data...")
+        time.sleep(5)
+
+        errors = []
+
+        # 3. Exclude one target from self.pool[1] while IOR is running.
+        excluded_pool_num = 1
+        non_excluded_pool_num = 0
+        self.log.info("Exclude rank 1 target 0")
+        self.pool[excluded_pool_num].exclude(ranks=[1], tgt_idx="0")
```

Comment:
SX, not S1. Sorry

Reply:
In case of SX, how are we identifying the rank chosen for IO is always rank1? If it is hardcoded in the actual daos code then it's fine. If it depends on some criteria , we should incorporate that logic.